File: M3Tools/Components/SelectReceipient.vb
Imports System.Windows.Forms

Public Class SelectReceipient
	Private __dialog As ReciepientSelection

	Private ReadOnly Property RecipientSelection As ReciepientSelection
		Get
			If __dialog Is Nothing Then
				__dialog = New ReciepientSelection
			ElseIf __dialog.IsDisposed Then
				__dialog = New ReciepientSelection
			End If

			Return __dialog
		End Get
	End Property

	Function ShowDialog() As DialogResult
		Return RecipientSelection.ShowDialog()
	End Function
End Class

File: M3Tools/Components/SelectReceipient.cs
using System.Windows.Forms;

namespace SPPBC.M3Tools
{

	/// <summary>
	/// 
	/// </summary>
    public partial class SelectReceipient
    {
        private ReciepientSelection __dialog;

        private ReciepientSelection RecipientSelection
        {
            get
            {
                if (__dialog is null)
                {
                    __dialog = new ReciepientSelection();
                }
                else if (__dialog.IsDisposed)
                {
                    __dialog = new ReciepientSelection();
                }

                return __dialog;
            }
        }

		/// <summary>
		/// Shows the selection dialog
		/// </summary>
		/// <returns></returns>
        public DialogResult ShowDialog()
        {
            return RecipientSelection.ShowDialog();
        }
    }
}
